### ISSUE TITLE
Function names `mb_tail_off`, `mb_head_off` and `utf_head_off` are not clear about how they treat composing characters

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#9222]
With UTF-8 encoding, `mb_tail_off` doesn't check if a character is a composing character, so it treats composing characters as separate characters. Meanwhile, `mb_head_off` is an alias to `utf_head_off`, which treats composing characters as a part of the preceding character. I'm not sure what `mb_tail_off` and `mb_head_off` can be renamed to, but I think `utf_head_off` can be renamed to `utfc_head_off` to reflect that it treats composing characters as a part of the preceding character (just like the difference between `utf_ptr2len` and `utfc_ptr2len`).

**Comment by brammool** [comment]
> With UTF-8 encoding, `mb_tail_off` doesn't check if a character is a
> composing character, so it treats composing characters as separate
> characters. Meanwhile, `mb_head_off` is an alias to `utf_head_off`,
> which treats composing characters as a part of the preceding
> character. I'm not sure what `mb_tail_off` and `mb_head_off` can be
> renamed to, but I think `utf_head_off` can be renamed to
> `utfc_head_off` to reflect that it treats composing characters as a
> part of the preceding character (just like the difference between
> `utf_ptr2len` and `utfc_ptr2len`).

Yes, this sometimes leads to mistakes.  Also, mb_tail_off is a function
while mb_head_off is a pointer to a function.

We should have a nice short name for "includes composing characters".
We currently have "utf_" and "utfc_", but that's not clear enough.

For example "mb_ptr2len" includes composing characters.  "mbc_ptr2len"
isn't different enough.  "mb_cps_ptr2len" perhaps?

-- 
SIGIRO -- irony detected (iron core dumped)

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by zeertzjq** [comment]
Also, things are a bit different when it comes to `mb_ptr2char_adv` and `mb_cptr2char_adv`. Here `mb_ptr2char_adv` ignores composing characters. whereas `mb_cptr2char_adv` returns them separately.
